config/routes.rb

```diff
@@ -82,9 +86,15 @@
       resources :campaigns, except: [:show]
       resources :emails, only: [:show, :update, :index]
       resources :physical_ticket, only: [:index]
-      resources :roles, except: [ :new, :create ] do
+      resources :roles, only: [:edit]
+      resources :roles, except: [ :new, :create, :edit ] do
         member do
           post :toggle_user
+          get ':track_name' => 'roles#show', as: 'track'
```

Comment:
What about removing all those extra lines and doing it in a more rails way? Perhaps nest the roles resources under the admin/tracks? Please think about how we could simply this route.